### PR TITLE
Improved map voting mechanics for admins.

### PR DESCRIPTION
**Pull request by vogelca**
A map vote from an admin will no longer bypass the percentage system
and force the map to change. It will be treated like a regular vote
instead.

If the need to force a change of map arises, there is a new command,
forcemap <mapname>, which does precisely that.

Only admins can execute forcemap, and it will refuse to work if a match
is currently in progress (admins can forcebreak the match first if
necessary).

### CODE REVIEW

**Comment by tcsabina** [comment]
Hey phylter!

Nice work!

Could you beautify the code a bit?
- variable declarations/definitions as first thing in a function (you know, the conventional C way ;))
- after a closing curly brace (block closing) add a newline
- add a newline before a return statement

Regards,
Toma

File: src/admin.c [hunk 1]
```diff
@@ -745,6 +745,34 @@ void AdminForceBreak()
...
+		return;
+	}
+	if (trap_CmdArgc() < 2)
+	{
+		G_sprint(self, 2, "Usage: forcemap %s\n", redtext("<mapname>"));
+		return;
+	}
+	char mapname[128];
+	trap_CmdArgv(1, mapname, sizeof(mapname));
+	extern int GetMapNum(char *);
```
Comment (tcsabina) [inline src/admin.c:766]: add the function declaration of GetMapNum() to g_local.h instead. There are already functions from maps.c there.

File: src/admin.c [hunk 2]
```diff
@@ -745,6 +745,34 @@ void AdminForceBreak()
...
+	if (GetMapNum(mapname) <= 0)
+	{
+		G_sprint(self, 2, "Map %s not available on this server.\n", redtext(mapname));
+		return;
```
Comment (tcsabina) [inline src/admin.c:770]: add newline before the 'return'

File: src/admin.c [hunk 3]
```diff
@@ -745,6 +745,34 @@ void AdminForceBreak()
...
+	}
```
Comment (tcsabina) [inline src/admin.c:771]: add newline after the closing brace

File: src/vote.c [hunk 4]
```diff
@@ -458,38 +458,27 @@ int vote_get_maps()
 
 void vote_check_map()
 {
-	int vt_req = get_votes_req( OV_MAP, true);
-	char *m = "";
-
-	if ((maps_voted_idx < 0) || strnull(m = GetMapName(maps_voted[maps_voted_idx].map_id)))
+	int vt_req = get_votes_req(OV_MAP, true);
```
Comment (tcsabina) [inline src/vote.c:461]: add a newline after the variable declaration/definition part is done

File: src/vote.c [hunk 5]
```diff
@@ -458,38 +458,27 @@ int vote_get_maps()
...
+	if (maps_voted_idx < 0)
 	{
 		return;
 	}
-
-	if (!k_matchLess)
+	char *mapname = GetMapName(maps_voted[maps_voted_idx].map_id);
```
Comment (tcsabina) [inline src/vote.c:466]: move this to the beginning of the function

File: src/vote.c [hunk 6]
```diff
@@ -458,38 +458,27 @@ int vote_get_maps()
...
+		return;
 	}
-	else if (!vt_req)
+	if (!k_matchLess && match_in_progress)
 	{
-		G_bprint(2, "%s votes for mapchange.\n", redtext("Majority"));
+		return;
 	}
-	else
+	if (vt_req)
```
Comment (tcsabina) [inline src/vote.c:475]: add a newline after the closing brace

File: src/vote.c [hunk 7]
```diff
@@ -458,38 +458,27 @@ int vote_get_maps()
...
-		}
-	}
-
-	if (maps_voted[maps_voted_idx].admins)
-	{
-		G_bprint(2, "%s\n", redtext("Admin veto"));
+		return;
 	}
-	else if (!vt_req)
+	if (!k_matchLess && match_in_progress)
```
Comment (tcsabina) [inline src/vote.c:471]: add a newline after the closing brace

File: src/vote.c [hunk 8]
```diff
@@ -458,38 +458,27 @@ int vote_get_maps()
...
-	else
+	if (vt_req)
 	{
 		return;
 	}
-
-	vote_clear( OV_MAP);
-
-	changelevel(m);
+	G_bprint(2, "%s votes for mapchange.\n", redtext("Majority"));
```
Comment (tcsabina) [inline src/vote.c:479]: add a newline after the closing brace

**Review by tcsabina — CHANGES_REQUESTED** [review]

**Comment by vogelca** [comment]
Beauty is subjective.

I have good reasons for doing things the way I do.

Nevertheless, I'll grant your request because
 1. It's the style used throughout the project;
 2. I always finish what I start.

But you have to promise one thing okay?

**Pull request merged**